src/repositories/movie_repository.go

```diff
@@ -1,42 +1,296 @@
-	// TODO implement
-	return nil, errors.New("not implemented")
+func (mr *MovieRepository) CreateMovie(movie *model.Movies) *models.KTSError {
+	// Create the insert statement
+	insertQuery := table.Movies.INSERT(
+		table.Movies.Title,
+		table.Movies.Description,
+		table.Movies.BannerPicURL,
+		table.Movies.CoverPicURL,
+		table.Movies.TrailerURL,
+		table.Movies.Rating,
+		table.Movies.ReleaseDate,
+		table.Movies.TimeInMin,
+		table.Movies.Fsk).
+		VALUES(movie.Title,
+			movie.Description,
+			movie.BannerPicURL,
+			movie.CoverPicURL,
+			movie.TrailerURL,
+			movie.Rating,
+			movie.ReleaseDate,
+			movie.TimeInMin,
+			movie.Fsk)
+
+	// Execute the query
+	rows, err := insertQuery.Exec(mr.DatabaseManager.GetDatabaseConnection())
+	if err != nil {
+		return kts_errors.KTS_INTERNAL_ERROR
+	}
+
+	rowsAff, err := rows.RowsAffected()
+	if err != nil {
+		return kts_errors.KTS_INTERNAL_ERROR
+	}
+
+	if rowsAff == 0 {
+		return kts_errors.KTS_MOVIE_NOT_FOUND
+	}
+
+	return nil
 }
 
-func (mr *MovieRepository) CreateMovie(movie *model.Movies) error {
-	// TODO: implement
-	return errors.New("not implemented")
+func (mr *MovieRepository) UpdateMovie(movie *model.Movies) *models.KTSError {
+	// Create the update statement
+	updateQuery := table.Movies.UPDATE().
+		SET(
+			table.Movies.Title.SET(mysql.String(movie.Title)),
+			table.Movies.Description.SET(mysql.String(movie.Description)),
+			table.Movies.BannerPicURL.SET(mysql.String(*movie.BannerPicURL)),
+			table.Movies.CoverPicURL.SET(mysql.String(*movie.CoverPicURL)),
+			table.Movies.TrailerURL.SET(mysql.String(*movie.TrailerURL)),
+			table.Movies.Rating.SET(mysql.Float(*movie.Rating)),
+			table.Movies.ReleaseDate.SET(mysql.DateT(movie.ReleaseDate)),
+			table.Movies.TimeInMin.SET(mysql.Int32(movie.TimeInMin)),
+			table.Movies.Fsk.SET(mysql.Int32(movie.Fsk)),
+		).WHERE(
+		table.Movies.ID.EQ(utils.MysqlUuid(movie.ID)),
+	)
+
+	// Execute the query
+	rows, err := updateQuery.Exec(mr.DatabaseManager.GetDatabaseConnection())
+	if err != nil {
+		return kts_errors.KTS_INTERNAL_ERROR
+	}
+
+	rowsAff, err := rows.RowsAffected()
+	if err != nil {
+		return kts_errors.KTS_INTERNAL_ERROR
+	}
+
+	if rowsAff == 0 {
+		return kts_errors.KTS_MOVIE_NOT_FOUND
+	}
+
+	return nil
+}
+
+func (mr *MovieRepository) DeleteMovie(movieId *uuid.UUID) *models.KTSError {
+	// Create the delete statement
+	deleteQuery := table.Movies.DELETE().WHERE(table.Movies.ID.EQ(utils.MysqlUuid(movieId)))
+
+	// Execute the query
+	rows, err := deleteQuery.Exec(mr.DatabaseManager.GetDatabaseConnection())
+	if err != nil {
+		return kts_errors.KTS_INTERNAL_ERROR
+	}
+
+	rowsAff, err := rows.RowsAffected()
+	if err != nil {
+		return kts_errors.KTS_INTERNAL_ERROR
+	}
+
+	if rowsAff == 0 {
+		return kts_errors.KTS_MOVIE_NOT_FOUND
+	}
+
+	return nil
+}
+
+// One Movie with all Genres
+func (mr *MovieRepository) GetMovieByIdWithGenre(movieId *uuid.UUID) (*models.MovieWithGenres, *models.KTSError) {
```

Comment:
Wouldnt extra write this, but simply rewrite the GetMovieBy id and add all needed attributes to there

Reply:
Removed